src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,105 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double CAPACITY_INDEX = 1.5;
+    private int arraySize;
+    private T[] items;
+
+    public ArrayList() {
+        items = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (arraySize == items.length) {
+            resize();
+        }
+        items[arraySize++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        checkIndexForAdd(index);
+        if (arraySize == items.length) {
+            resize();
+        }
+        System.arraycopy(items, index, items, index + 1, arraySize++ - index);
+        items[index] = value;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return items[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        checkIndex(index);
+        items[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        T deletedObject = items[index];
+        System.arraycopy(items, index + 1, items, index, --arraySize - index);
+        return deletedObject;
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        for (int i = 0; i < arraySize; i++) {
+            if (element == null) {
+                if (items[i] == null) {
+                    return remove(i);
+                }
+            } else {
+                if (element.equals(items[i])) {
+                    return remove(i);
+                }
+            }
```

Comment:
actual logic is the same in both clauses of if-else - it means you don't need it, just use correct null save equality checker in condition